xml_feed/xml_feed.py

```diff
@@ -110,17 +148,31 @@ def trigger(self, previous_result, config, names, tags, run_all, rate_limit,
         if not config['collection'].endswith('/'):
             config['collection'] = '{}/'.format(config['collection'])
 
-        config = self.parse_config(config)
+        options = Options(config)
+
+        sanitized_config = dict(
+            (k,v) for k,v in config.iteritems()
+            if k not in CONFIG_FIELDS_TO_REMOVE)
+        config_message = self.parse_config(sanitized_config)
 
-        raw_feed = self._download_feed(config.url)
-        for article in self._parse_articles(raw_feed):
-            pod_path = '{}{}.html'.format(config.collection, article.slug)
+        raw_feed = self._download_feed(config_message.url)
+        for article in self._parse_articles(raw_feed, options):
+            safe_article_slug = article.slug.replace('.', '-dot')
```

Comment:
Why do you need to replace the slugs?

Reply:
I was receiving the following stack-trace indicating that repeated dots were a no-go for file paths in Grow, though perhaps the better spot to make a change would be in pods.py to allow them for file-names but not path pieces? What do you think?

[12:17:09] Saving /content/medium_feed/wing-selected-to-participate-in-the-faa's-uas-traffic-management-pilot-program-as-part-of-the....html
Traceback (most recent call last):
  File "grow", line 76, in <module>
  File "site-packages/click/core.py", line 722, in __call__
  File "site-packages/click/core.py", line 697, in main
  File "site-packages/click/core.py", line 1066, in invoke
  File "site-packages/click/core.py", line 895, in invoke
  File "site-packages/click/core.py", line 535, in invoke
  File "grow/commands/subcommands/preprocess.py", line 42, in preprocess
  File "grow/pods/pods.py", line 773, in preprocess
  File "grow/extensions/extension_controller.py", line 50, in trigger
  File "grow/extensions/hook_controller.py", line 28, in trigger
  File "/Users/amacdonald/grow-projects/google-wing/extensions/xml_feed/xml_feed.py", line 188, in trigger
    self.pod.write_file(pod_path, raw_content)
  File "grow/pods/pods.py", line 884, in write_file
  File "grow/pods/pods.py", line 153, in _normalize_path
ValueError: .. not allowed in file paths.
[41261] Failed to execute script grow

Reply:
How about converting all the repeating dots into singular dots? We could do a regex replace all repeating periods (`[\.]{2,}`) with a singular period.

Reply:
Sounds good, I switched it out to remove the duplicates, and remove any trailing dot since we add ".html" afterwards.